src/Servers/Kestrel/Core/src/ListenOptionsHttpsExtensions.cs

```diff
@@ -219,6 +219,8 @@ public static ListenOptions UseHttps(this ListenOptions listenOptions, HttpsConn
             var loggerFactory = listenOptions.KestrelServerOptions?.ApplicationServices.GetRequiredService<ILoggerFactory>() ?? NullLoggerFactory.Instance;
 
             listenOptions.IsTls = true;
+            listenOptions.HttpsOptions = httpsOptions;
```

Comment:
Not all of the UseHttps overloads share this code path.
https://github.com/dotnet/aspnetcore/blob/0eb496acb249164668e6d889c7e8d3fe4d38da1b/src/Servers/Kestrel/Core/src/ListenOptionsHttpsExtensions.cs#L243

MSQuic doesn't support Per-SNI callbacks, right? How do we communicate that at config time?

Reply:
I think we need to figure this out.

cc @geoffkizer

Reply:
If per-SNI callbacks aren't supported then it is a simple matter to throw an error if `ServerCertificate` is null. Should raise with msquic that it is something we'd want in the future.

If they are supported then we'll need to get the details from msquic and plum support for it through the various layers.

Reply:
> MSQuic doesn't support Per-SNI callbacks, right? How do we communicate that at config time?

Clarify what you mean here? Do you mean selecting a certificate based on SNI? I believe MsQuic does support this, we just don't currently support it in System.Net.Quic.

Reply:
That's a start, but in 5.0 SslStream added the ability to specify all TLS settings per SNI, not just the certificate. That's the overload I linked above.

Reply:
There are a number of callbacks in our TLS support. I'm not experience in this area so I'd like to come up a list of what they're each called, how important it is to support them, and whether msquic can support them today.

Here is our list of UseHttps methods:

https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.hosting.listenoptionshttpsextensions?view=aspnetcore-5.0

There is UseHttps overload with a ServerSelectionCallback - https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.hosting.listenoptionshttpsextensions.usehttps?view=aspnetcore-5.0#Microsoft_AspNetCore_Hosting_ListenOptionsHttpsExtensions_UseHttps_Microsoft_AspNetCore_Server_Kestrel_Core_ListenOptions_System_Net_Security_ServerOptionsSelectionCallback_System_Object_

`public delegate ValueTask<SslServerAuthenticationOptions> ServerOptionsSelectionCallback(SslStream stream, SslClientHelloInfo clientHelloInfo, object? state, CancellationToken cancellationToken);`

Then there is the ServerCertificateSelector callback on the Https options type. https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/endpoints?view=aspnetcore-5.0#configure-endpoints-using-server-name-indication